build/documentation/copy-package-readme.js

```diff
@@ -29,36 +31,130 @@ process.argv.forEach(function (val, index) {
         console.log('Invalid argument used. To perform a dry-run use --dry-run');
         process.exit(1);
     }
+
 });
 
 /**
- * Function to copy readme files to the docs/en/packages folder
+ * Run script
+ * Based off presence of --dry-run parameter
  */
-function copyReadmeFiles() {
+dryRun? runDryRun() : copyReadmeFiles();
+
+
+/**
+ * Function that describes what this script will do and
+ * where, without making any actual changes
+ */
+function runDryRun() {
 
     const resolvedSrcReadmePaths = path.resolve(rootDir, srcReadmePaths);
 
-    glob(resolvedSrcReadmePaths, {realpath:true}, function(error, files) {
-        
-        files.forEach((filePath) => {
-            const destReadmePath = filePath.replace(/(\bpackages\b)(?!.*\1)/, destDir);
+    console.log("   In docs/en/packages/, this script would...");
+
+    glob(resolvedSrcReadmePaths, {realpath:true}, function(error, srcFiles) {
+        srcFiles.forEach((srcReadmePath) => {    
+
+            const destReadmePath = srcReadmePath.replace(/(\bpackages\b)(?!.*\1)/, destDir);
             const dirPath = path.dirname(destReadmePath);
+            const srcDirPath = path.dirname(srcReadmePath);
+        
+            var title = srcDirPath
+                .split(path.sep)
+                .pop()
+        
+            var msg = '';
+
             if (!fs.existsSync(dirPath)) {
-                dryRun ? console.log("----> Would create folder " + dirPath) : fs.mkdirSync(dirPath);
+                msg += "      ...CREATE folder '" + title + "'\n";
```

Comment:
There is a lot of string manipulation given the requirements.  I suggestion using JavaScript console.log string formatters for better maintainability.

Reply:
We should consider a testing strategy for the documentation scripts and if it makes sense a logging utility script.

Reply:
I agree, we will add a testing strategy after when we rip out Docusaurus to replace with custom React app. I also have on the backlog to investigate logging libraries across all FAST-DNA. But, neither are required as part of #1312.